### ISSUE TITLE
Source.actorPublisher[T](Props[A]) results in Exception  

### ISSUE THREAD

**Issue by solomono** [unicredit/akka.js#38]
To reproduce change the last line in the Run.run in the stream demo class to 
    `Streams.complexFlow(true)`

This causes the  val toStringActorPublisherSource in method Streams.complexFlow to be assigned to the expression  `Source.actorPublisher[String](Props[ToStringActorPublisher])`

Exception stack trace below 

`ERROR] [Tue Jul 22:13:42 GMT+0100 2016] [main] [akka://streams/user/StreamSupervisor-1/flow-1-1-actorPublisherSource] null
akka.actor.ActorInitializationException: akka://streams/user/StreamSupervisor-1/flow-1-1-actorPublisherSource: exception during creation
  at $c_Lakka_actor_ActorInitializationException.$c_jl_Throwable.fillInStackTrace__jl_Throwable(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:27427:14)
  at $c_Lakka_actor_ActorInitializationException.$c_jl_Throwable.init___T__jl_Throwable(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:27516:8)
  at akka.actor.ActorInitializationException.<init>(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:63826:52)
  at akka.actor.ActorCell.create(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:78844:69)
  at akka.actor.ActorCell.invokeAll$1(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:78895:18)
  at $c_Lakka_dispatch_Dispatcher$$anon$1.$c_Lakka_dispatch_Mailbox.processAllSystemMessages__V(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30826:12)
  at $c_Lakka_dispatch_Dispatcher$$anon$1.$c_Lakka_dispatch_Mailbox.run__V(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30776:12)
  at null._onTimeout(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30653:19)
  at Timer.listOnTimeout(timers.js:119:15)
Caused by: TypeError: Cannot read property 'prototype' of undefined
  at $newJSObjectWithVarargs(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:449:45)
  at akka.util.Reflect$.instantiate(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:13835:12)
  at akka.util.Reflect$.instantiate(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:13848:17)
  at akka.actor.NoArgsReflectConstructor.produce(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:21589:57)
  at akka.actor.ActorCell.newActor(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:79009:73)
  at akka.actor.ActorCell.create(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:78814:24)
  at akka.actor.ActorCell.invokeAll$1(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:78895:18)
  at $c_Lakka_dispatch_Dispatcher$$anon$1.$c_Lakka_dispatch_Mailbox.processAllSystemMessages__V(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30826:12)
  at $c_Lakka_dispatch_Dispatcher$$anon$1.$c_Lakka_dispatch_Mailbox.run__V(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30776:12)
  at null._onTimeout(/Users/solomono/Development/Repository/Git/akka.js-fork/akka-stream-demo/js/target/scala-2.11/demo-fastopt.js:30653:19)`

**Comment by andreaTP** [comment]
I have your example properly working :-) I checked against it ...
Probably since now everything is asynchronous  you need to check that the system is already started before you create the stream:

``` scala
import system.dispatcher
import scala.concurrent.duration._
system.scheduler.scheduleOnce(0 millis){
  //your code here
}
```

this is a known limitation/bug/whatever I must document it properly at some point....

**Comment by andreaTP** [comment]
b.t.w. I cannot reproduce your behavior, please PR the code on the examples project on github and we can be aligned :-) thanks in advance!

**Comment by solomono** [comment]
When Streams.complexFlow(false)
We see the two lines for each index

```
Foreach Sink: Wed Jul 12:55:02 GMT+0100 2016 - factorial(1) = 1
Actor Sink  : Wed Jul 12:55:02 GMT+0100 2016 - factorial(1) = 1
Actor Sink  : Wed Jul 12:55:03 GMT+0100 2016 - factorial(2) = 1
Foreach Sink: Wed Jul 12:55:03 GMT+0100 2016 - factorial(2) = 1
Foreach Sink: Wed Jul 12:55:04 GMT+0100 2016 - factorial(3) = 2
Actor Sink  : Wed Jul 12:55:04 GMT+0100 2016 - factorial(3) = 2
Foreach Sink: Wed Jul 12:55:05 GMT+0100 2016 - factorial(4) = 6
Actor Sink  : Wed Jul 12:55:05 GMT+0100 2016 - factorial(4) = 6
.
.
```

When Streams.complexFlow(true)
We see ONLY one  line for each index

```
Foreach Sink: Wed Jul 12:46:40 GMT+0100 2016 - factorial(1) = 1
Foreach Sink: Wed Jul 12:46:41 GMT+0100 2016 - factorial(2) = 1
Foreach Sink: Wed Jul 12:46:42 GMT+0100 2016 - factorial(3) = 2
Foreach Sink: Wed Jul 12:46:43 GMT+0100 2016 - factorial(4) = 6
.
.
```

You can see the stack trace just above the output of the stream in the second case

**Comment by andreaTP** [comment]
fixed on branch `master` of the example, creation of actors with the Props"type" signature is supported under certain constraints, I have to detail this at some point ...
Unfortunately on this I need to first integrate a Scala.js linker plugin.

Thanks!

**Comment by solomono** [comment]
As you stated it is not this call in particular, but using Props.apply that takes just the type parameter .
I changed the code to   `Source.actorPublisher[String](Props(new ToStringActorPublisher()))`, and it works.
Thanks

**Comment by solomono** [comment]
Reviewing the closed issues. I see that the stack trace reported for closed issue #10  is exactly the same as I obtain here using Props.apply that takes just the type parameter.

**Comment by andreaTP** [comment]
great!
